connectionmanager/connectionmanager.go

```diff
@@ -93,7 +93,8 @@ func connect(
 	setReconnecting(proto, true)
 	for IsReconnecting(proto) {
 		i++
-		c, err = connection.NewConnection(u, token, writeWait, maxMessageSize, DefaultPingWait)
+		c, err = connection.NewConnection(ctx, u, token,
+			writeWait, maxMessageSize, DefaultPingWait)
```

Comment:
I verified that this is the correct place to introduce the back-off. I also verified that we need it in this (very elegant) form or the other PR (also not bad).

Reply:
Oops... I forgot to push my latest amendments to this PR.